pkg/provisioner/cloudprovisioner.go

```diff
@@ -329,36 +329,39 @@ func (c *CloudProvisioner) ListVolumes(
 	return c.listVolumesInNodeResourceGroup(ctx, start, int(maxEntries))
 }
 
+// PublishVolume calls AttachDisk asynchronously and returns early lun assignment value and a channel for the async attach results.
 func (c *CloudProvisioner) PublishVolume(
 	ctx context.Context,
 	volumeID string,
 	nodeID string,
-	volumeContext map[string]string) (map[string]string, error) {
+	volumeContext map[string]string) (chan azureutils.CloudAttachResult, map[string]string, error) {
 	var err error
+	var attachResultCh chan azureutils.CloudAttachResult
```

Comment:
`CloudAttachResult` should just be part of the provisioner package.

Reply:
Shouldn't we keep the type definition in a separate file for interfaces?

Reply:
The result is specific to the `CloudProvisioner`. I don't believe we have a use case for it otherwise and it's OK to have type dependencies on the `provisioner` package event when interfaces are defined in another package.